### ISSUE TITLE
object 'Cagsurv5' not found when calling plot.flexsurvreg

### ISSUE THREAD

**Issue by jrdnmdhl** [chjackson/flexsurv-dev#32]
The development version of the heemod package is currently failing on travis CI due to a fairly innocuous call to plot a flexsurvreg object which results in an error:
https://travis-ci.org/pierucci/heemod/jobs/213457178

```r
library(flexsurv)

fit_w <- flexsurvreg(
  formula = Surv(futime, fustat) ~ 1,
  data = ovarian, dist = "weibull"
)
plot(fit_w)
fit_spl <- flexsurvspline(
  formula = Surv(futime, fustat) ~ 1,
  data = ovarian,
  scale = "odds",
  k=1
)
plot(fit_spl)
```

I haven't been able to reproduce this yet on my machine.  It has so far only been found on: `R Under development (unstable) (2017-03-21 r72380)`.

I believe the error itself occurs within the survival package, but it isn't clear to me whether the problem arises from the survival package or the way it is being called from plot.flexsurvreg.

**Comment by MattWiener** [comment]
Still an issue in r72450

**Comment by chjackson** [comment]
I can reproduce this under 2017-03-26 r72404, but the error comes from the flexsurvspline fit, not the plot.flexsurvreg call.  I found unpredictable errors where R functions from external packages failed to find their own packages' C routines when called from flexsurv. 

I recently added registration of native routines, as recommended here, http://r.789695.n4.nabble.com/Registration-of-native-routines-td4728874.html in order to pass CRAN check.

In the latest commit, I have changed this slightly to allow dynamic symbols, and these errors have gone away for me. https://github.com/chjackson/flexsurv-dev/commit/efd9d76129f83c58d8775112f21659584760763c#diff-7b61679bcc91478cdb6dbc87f9d8ba80

But I'm quite confused by this.  "Writing R Extensions" documents this as "The R_useDynamicSymbols call says the DLL is not to be searched for entry points specified by character strings so .C etc calls will only find registered symbols".    Why should this setting affect whether other packages can find their own C routines when called from my package? 

And why did it pass the checks on CRAN's machines when I released it to CRAN a few days ago?? https://cran.r-project.org/web/checks/check_results_flexsurv.html
